Apps/WebClient/src/NewClientApp/src/components/services/OrganDonorDetailsCard.vue

```diff
@@ -0,0 +1,170 @@
+﻿<script setup lang="ts">
+import saveAs from "file-saver";
+import { computed, ref } from "vue";
+
+import MessageModalComponent from "@/components/shared/MessageModalComponent.vue";
+import {
+    OrganDonorRegistration,
+    PatientData,
+    PatientDataFile,
+    PatientDataType,
+} from "@/models/patientDataResponse";
+import { ILogger } from "@/services/interfaces";
+import SnowPlow from "@/utility/snowPlow";
+import { container } from "@/ioc/container";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import { usePatientDataStore } from "@/stores/patientData";
+import HgCardComponent from "@/components/shared/HgCardComponent.vue";
+import HgButtonComponent from "@/components/shared/HgButtonComponent.vue";
+
+interface Props {
+    hdid: string;
+}
+const props = defineProps<Props>();
+
+const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+const patientDataStore = usePatientDataStore();
+
+const sensitiveDocumentModal =
+    ref<InstanceType<typeof MessageModalComponent>>();
+
+const patientData = computed<PatientData[]>(() => {
+    return patientDataStore.patientData(props.hdid, [
+        PatientDataType.OrganDonorRegistrationStatus,
+    ]);
+});
+
+const isLoadingFile = computed<boolean>(
+    () =>
+        registrationData.value?.registrationFileId !== undefined &&
+        patientDataStore.isPatientDataFileLoading(
+            registrationData.value?.registrationFileId
+        )
+);
+
+const registrationData = computed<OrganDonorRegistration | undefined>(() => {
+    return patientData.value
+        ? (patientData.value[0] as OrganDonorRegistration)
+        : undefined;
+});
+
+function getDecisionFile(): void {
+    const registrationDataValue = registrationData.value;
+    if (registrationDataValue && registrationDataValue.registrationFileId) {
+        SnowPlow.trackEvent({
+            action: "download_report",
+            text: "Organ Donor",
+        });
+        patientDataStore
+            .retrievePatientDataFile(
+                props.hdid,
+                registrationDataValue.registrationFileId
+            )
+            .then(
+                (patientFile: PatientDataFile) =>
+                    new Blob([new Uint8Array(patientFile.content)], {
+                        type: patientFile.contentType,
+                    })
+            )
+            .then((blob) => saveAs(blob, `Organ_Donor_Registration.pdf`))
+            .catch((err) => logger.error(err));
+    }
+}
+
+function showConfirmationModal(): void {
+    sensitiveDocumentModal.value?.showModal();
+}
+</script>
```

Comment:
There used to be code here that added a custom error when this service was blocked. Is that handled differently now?

Reply:
I see that code was from a recent change in the dev branch.  I've now ported those changes to this branch.